tests/integration/test_application.py

```diff
@@ -269,15 +269,31 @@ def test_invalid_command_argument(monkeypatch, capsys, app):
 
 def test_global_environment(
     create_app,
+    mocker,
     monkeypatch,
     tmp_path,
 ):
     """Test that the global environment is correctly populated during the build process."""
     monkeypatch.chdir(tmp_path)
     shutil.copytree(VALID_PROJECTS_DIR / "environment", tmp_path, dirs_exist_ok=True)
 
+    # a build plan that builds for s390x (a cross-compiling scenario unless on s390x)
+    mocker.patch(
+        "tests.conftest.MyBuildPlanner.get_build_plan",
+        return_value=[
+            models.BuildInfo(
+                platform="s390x",
+                build_on=util.get_host_architecture(),
+                build_for="s390x",
+                base=util.get_host_base(),
+            ),
+        ],
+    )
+
     # Run in destructive mode
-    monkeypatch.setattr("sys.argv", ["testcraft", "prime", "--destructive-mode"])
+    monkeypatch.setattr(
+        "sys.argv", ["testcraft", "prime", "--destructive-mode", "--build-for", "s390x"]
```

Comment:
taking into account my previous (big) comment, this test should work both with and without `--build-for s390x`